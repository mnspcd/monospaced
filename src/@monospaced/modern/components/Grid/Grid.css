.mdrn .Grid {
  display: grid;
  grid-gap: 1.5em 0.75em;
  grid-template-columns: 1fr;
  margin: 0 auto;
  max-width: 36.25em;
}

@supports (display: grid) {
  .mdrn .Grid {
    max-width: 56.75em;
  }
}

@media (--larger-viewport) {
  .mdrn .Grid {
    grid-gap: 0.75em;
    grid-template-columns: repeat(12, 1fr);
  }
}

.mdrn .Grid-item {
  margin-bottom: 1.5em;
}

.mdrn .Grid-item:last-child {
  margin-bottom: 0;
}

@supports (display: grid) {
  .mdrn .Grid-item {
    margin-bottom: 0;
  }
}

@media (--smaller-viewport) {
  .mdrn .Grid-item {
    grid-column-end: auto !important;
    grid-column-start: auto !important;
    grid-row-end: auto !important;
    grid-row-start: auto !important;
  }
}
