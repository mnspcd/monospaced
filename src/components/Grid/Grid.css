.Grid {
  margin: 0 auto;
  max-width: 56.75em;
  padding-left: 1em;
  padding-right: 1em;
}

@media (--larger-viewport) {
  .Grid {
    display: grid;
    grid-gap: 0.75em;
    grid-template-columns: repeat(6, 1fr);
  }
}

.Grid-item {
  margin-bottom: 0.75em;
}

@media (--larger-viewport) {
  .Grid-item {
    margin-bottom: 0;
  }
}

.Grid-item:last-child {
  margin-bottom: 0;
}
