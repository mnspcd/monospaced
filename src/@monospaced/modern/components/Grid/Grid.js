import PropTypes from "prop-types";
import React from "react";

import "./Grid.css";

/**
 * Use `Grid` to layout content in a 12-column grid.
 */
const Grid = ({ children, padding = "0 1em" }) => (
  <div className="Grid" style={{ padding }}>
    {children}
  </div>
);

Grid.alignments = ["center", "end", "start"];

Grid.propTypes = {
  /**
   * Grid.Item components to display in the Grid.
   */
  children: PropTypes.node.isRequired,

  /**
   * A valid CSS padding value for the Grid.
   */
  padding: PropTypes.string,
};

Grid.Item = ({
  align,
  children,
  colSpan,
  colStart,
  justify,
  rowSpan,
  rowStart,
}) => (
  <div
    className="Grid-item"
    style={{
      alignSelf: align,
      gridColumnEnd: colSpan && `span ${colSpan}`,
      gridColumnStart: colStart,
      gridRowEnd: rowSpan && `span ${rowSpan}`,
      gridRowStart: rowStart,
      justifySelf: justify,
    }}
  >
    {children}
  </div>
);

Grid.Item.displayName = "Grid.Item";

Grid.Item.propTypes = {
  align: PropTypes.oneOf(Grid.alignments),
  children: PropTypes.node,
  colSpan: PropTypes.string,
  colStart: PropTypes.string,
  justify: PropTypes.oneOf(Grid.alignments),
  rowSpan: PropTypes.string,
  rowStart: PropTypes.string,
};

export default Grid;
