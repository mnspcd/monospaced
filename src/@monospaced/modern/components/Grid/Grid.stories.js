import { number, select, text } from "@storybook/addon-knobs";
import React from "react";

import Block from "../Block";

import Grid, { Item } from "./Grid";

export default {
  component: Grid,
  parameters: {
    subcomponents: {
      "Grid.Item": Item,
    },
  },
  title: "Components/Grid",
};

export const basic = () => (
  <Grid padding="0">
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item>
      <Block height="4.5em" />
    </Grid.Item>
  </Grid>
);

export const centered = () => (
  <Grid padding="0">
    <Grid.Item colSpan="8" colStart="3">
      <Block height="4.5em" />
    </Grid.Item>
  </Grid>
);

export const sideNote = () => (
  <Grid>
    <Grid.Item align="center" colSpan="2">
      <Block />
    </Grid.Item>
    <Grid.Item colSpan="8">
      <Block height="4.5em" />
    </Grid.Item>
  </Grid>
);

export const constrained = () => (
  <Grid padding="0">
    <Grid.Item colSpan="10">
      <Block height="4.5em" />
    </Grid.Item>
  </Grid>
);

export const sideBar = () => (
  <Grid padding="0">
    <Grid.Item colSpan="4">
      <Block height="4.5em" />
    </Grid.Item>
    <Grid.Item colSpan="8">
      <Block height="4.5em" />
    </Grid.Item>
  </Grid>
);

export const knobs = () => {
  const numberOfItems = number(
    "number of items",
    12,
    {
      range: true,
      min: 1,
      max: 24,
    },
    "Story",
  );

  return (
    <Grid padding={text("padding", "0 1em", "Grid")}>
      <Grid.Item
        align={
          select("align", [""].concat(Grid.alignments), "", "Grid.Item") || null
        }
        colSpan={text("colSpan", "", "Grid.Item")}
        colStart={text("colStart", "", "Grid.Item")}
        justify={
          select("justify", [""].concat(Grid.alignments), "", "Grid.Item") ||
          null
        }
        rowSpan={text("rowSpan", "", "Grid.Item")}
        rowStart={text("rowStart", "", "Grid.Item")}
      >
        <Block></Block>
      </Grid.Item>
      {[...Array(numberOfItems - 1).keys()].map(i => (
        <Grid.Item key={i}>
          <Block height="4.5em" />
        </Grid.Item>
      ))}
    </Grid>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
