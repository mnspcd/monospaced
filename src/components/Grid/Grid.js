import PropTypes from "prop-types";
import React from "react";

import "./Grid.css";

const Grid = ({ children, rows }) => (
  <div
    className="Grid"
    style={{ gridTemplateRows: rows && `repeat(${rows}, 1fr)` }}
  >
    {children}
  </div>
);

Grid.alignments = ["center", "end", "start"];

Grid.propTypes = {
  children: PropTypes.node.isRequired,
  rows: PropTypes.string,
};

Grid.Item = ({
  align,
  children,
  colSpan,
  colStart,
  justify,
  rowSpan,
  rowStart,
}) => (
  <div
    className="Grid-item"
    style={{
      alignSelf: align,
      gridColumnEnd: colSpan && `span ${colSpan}`,
      gridColumnStart: colStart,
      gridRowEnd: rowSpan && `span ${rowSpan}`,
      gridRowStart: rowStart,
      justifySelf: justify,
    }}
  >
    {children}
  </div>
);

Grid.Item.displayName = "Grid.Item";

Grid.Item.propTypes = {
  align: PropTypes.oneOf(Grid.alignments),
  children: PropTypes.node,
  colSpan: PropTypes.string,
  colStart: PropTypes.string,
  justify: PropTypes.oneOf(Grid.alignments),
  rowSpan: PropTypes.string,
  rowStart: PropTypes.string,
};

export default Grid;
