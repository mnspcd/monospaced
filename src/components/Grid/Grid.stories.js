import React from "react";
import { number, select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Grid from "./Grid";
import Block from "../Block";

storiesOf("Components/Grid", module).add("Component", () => {
  const numberOfItems = number(
    "number of items",
    6,
    {
      range: true,
      min: 1,
      max: 24,
    },
    "Story",
  );

  return (
    <Grid>
      <Grid.Item
        align={
          select("align", [""].concat(Grid.alignments), "", "Grid.Item") || null
        }
        colSpan={text("colSpan", "", "Grid.Item")}
        colStart={text("colStart", "", "Grid.Item")}
        justify={
          select("justify", [""].concat(Grid.alignments), "", "Grid.Item") ||
          null
        }
        rowSpan={text("rowSpan", "", "Grid.Item")}
        rowStart={text("rowStart", "", "Grid.Item")}
      >
        <Block>Item</Block>
      </Grid.Item>
      {[...Array(numberOfItems - 1).keys()].map(i => (
        <Grid.Item key={i}>
          <Block height="6em" />
        </Grid.Item>
      ))}
    </Grid>
  );
});
