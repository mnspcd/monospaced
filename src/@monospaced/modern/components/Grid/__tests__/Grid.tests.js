/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Grid from "..";

describe("Grid component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item>children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render alignment correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item align="center">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render colSpan correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item colSpan="2">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render colStart correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item colStart="2">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render justification correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item justify="center">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render rowSpan correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item rowSpan="2">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render rowStart correctly", () => {
    const { container } = render(
      <Grid>
        <Grid.Item rowStart="2">children</Grid.Item>
      </Grid>,
    );

    expect(container).toMatchSnapshot();
  });
});
