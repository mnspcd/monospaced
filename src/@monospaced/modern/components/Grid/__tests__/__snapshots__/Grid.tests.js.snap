// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render alignment correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="align-self:center"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render colSpan correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="grid-column-end:span 2"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render colStart correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="grid-column-start:2"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render justification correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="justify-self:center"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render rowSpan correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="grid-row-end:span 2"
  >
    children
  </div>
</div>
`;

exports[`Grid component should render rowStart correctly 1`] = `
<div
  class="Grid"
  style="padding:0 1em"
>
  <div
    class="Grid-item"
    style="grid-row-start:2"
  >
    children
  </div>
</div>
`;
